soundata/datasets/singapura.py

```diff
@@ -0,0 +1,325 @@
+        *K. Ooi, K. N. Watcharasupat, S. Peksi, F. A. Karnapi, Z.-T. Ong, D. Chua, H.-W. Leow, L.-L. Kwok, X.-L. Ng, Z.-A. Loh, W.-S. Gan, "A Strongly-Labelled Polyphonic Dataset of Urban Sounds with Spatiotemporal Context," in Proceedings of the 13th Asia Pacific Signal and Information Processing Association Annual Summit and Conference, 2021.
+        
+    *License:*
+        Creative Commons Attribution-ShareAlike 4.0 International.
+
+"""
+import os
+from typing import Dict, List, Union
+
+
+import librosa
+import numpy as np
+import pandas as pd
+
+from soundata import download_utils, jams_utils, core, annotations, io
+
+# -- Add any relevant citations here
+BIBTEX = """
+@inproceedings{ooi2021singapura,
+    author    = "K. Ooi and K. N. Watcharasupat and S. Peksi and F. A. Karnapi and Z.-T. Ong and D. Chua and H.-W. Leow and L.-L. Kwok and X.-L. Ng and Z.-A. Loh and W.-S. Gan",
+    title     = "A Strongly-Labelled Polyphonic Dataset of Urban Sounds with Spatiotemporal Context",
+    booktitle = "Proceedings of the 13th Asia Pacific Signal and Information Processing Association Annual Summit and Conference",
+    location  = "Tokyo, Japan"
+    year      = 2021
+}
+"""
+
+# -- REMOTES is a dictionary containing all files that need to be downloaded.
+# -- The keys should be descriptive (e.g. 'annotations', 'audio').
+# -- When having data that can be partially downloaded, remember to set up
+# -- correctly destination_dir to download the files following the correct structure.
+
+meta_files = [
+    ("metadata", "labelled_metadata_public.csv", "c5beb6374e55abfe7cd50f4f498c8376"),
+    (
+        "labels",
+        "labels_public.zip",
+        "535242cf1094d95d086fc574874e9ddf",
+    ),  # the base zip should be the last file
+]
+
+audio_files = [
+    ("labelled.zip", "bdd46cc5e9187e97c37989b3b73e786e"),
+    ("labelled.z01", "98477daca861c6950cc8b620cecc286d"),
+    ("labelled.z02", "873b26cfe25bb3084e39e5af0dfebcad"),
+    ("labelled.z03", "71322a5c3ba33badfdd8e25c9ebf559a"),
+    ("labelled.z04", "a6d087babea1f797af99b81cb7c7ea4a"),
+]
+
+meta_remotes = {
+    k: download_utils.RemoteFileMetadata(
+        filename=f,
+        url=f"https://zenodo.org/record/5645825/files/{f}?download=1",
+        checksum=m,  # -- the md5 checksum
+        destination_dir=None,  # -- relative path for where to unzip the data, or None
+    )
+    for k, f, m in meta_files
+}
+
+# put as list for multipart zip
+audio_remotes = {
+    "audio": [
+        download_utils.RemoteFileMetadata(
+            filename=f,
+            url=f"https://zenodo.org/record/5645825/files/{f}?download=1",
+            checksum=m,  # -- the md5 checksum
+            destination_dir=None,  # -- relative path for where to unzip the data, or None
+        )
+        for f, m in audio_files
+    ]
+}
+
+REMOTES: Dict[
+    str,
+    Union[List[download_utils.RemoteFileMetadata], download_utils.RemoteFileMetadata],
+] = {**audio_remotes, **meta_remotes}
```

Comment:
The type annotation was added to prevent mypy from throwing error actually. mypy doesn't know how to merge two dicts with different element types so it throws error during the CI.